src/Dialect/ONNX/Transforms/Recompose.cpp

```diff
@@ -602,6 +602,57 @@ struct RecomposeQLinearMatMulFromQuantizeLinearPattern
   }
 };
 
+/// Merges nested ONNXConcatOps
+struct RecomposeConcatPattern : public OpRewritePattern<ONNXConcatOp> {
+  using OpRewritePattern<ONNXConcatOp>::OpRewritePattern;
+
+  // Helper function to check if an input is a mergeable Concat.
+  static bool isMergeableConcat(Value input, int64_t axis) {
+    auto innerConcat = input.getDefiningOp<ONNXConcatOp>();
```

Comment:
Please use concrete type here. I think it's `ONNXConcatOp innerConcat ...`. We avoid using `auto` as much as possible.